data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionException.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+/**
+ * @since 1.3
+ * Exception thrown by {@link ParseTreeCoercionService} methods.
+ */
+public class ParseTreeCoercionException extends Exception {
```

Comment:
From a caller's perspective, the parse tree doesn't mean much. I think we could use a better name here. Perhaps `ExpressionCoercionException`.